### PR TITLE
xds: add test for LRS retry and re-send load reports after stream closed

### PR DESCRIPTION
**Pull request by voidzcy**
This is a test case that covers LRS client re-establishes load reporting (send a new initial request and receives new LRS response which contains new load reporting interval) after the LB stream is closed.

The slight difference between this test case and `lrsStreamClosedAndRetried` is that it focuses on load reporting being back to normal after retry while the other focuses on re-establishing the load reporting RPC.

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java [hunk 1]
```diff
@@ -453,6 +453,86 @@ public void lrsStreamClosedAndRetried() {
...
+
+    // First LRS request sent.
+    verify(requestObserver).onNext(EXPECTED_INITIAL_REQ);
+    assertThat(logs).containsExactly("DEBUG: Initial LRS request sent: " + EXPECTED_INITIAL_REQ);
+    logs.poll();
+    assertEquals(0, fakeClock.numPendingTasks(LRS_RPC_RETRY_TASK_FILTER));
+
+    // Balancer sends back a normal response.
+    responseObserver.onNext(buildLrsResponse(100));
+    assertThat(logs).containsExactly(
```
Comment (dapengzhang0) [inline xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Verifying logs seems weird. Further more, `containsExactly()` is easily broken in the future and should not be relied on. Seems you don't need verify this.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java [hunk 2]
```diff
@@ -453,6 +453,86 @@ public void lrsStreamClosedAndRetried() {
...
+    verifyNoMoreInteractions(requestObserver);
+
+    // Balancer closes the stream with error.
+    responseObserver.onError(Status.UNKNOWN.asException());
+
+    // The unsent load report is cancelled.
+    assertEquals(0, fakeClock.numPendingTasks(LOAD_REPORTING_TASK_FILTER));
+    // Will retry immediately as balancer has responded previously.
+    verify(mockLoadReportingService, times(2)).streamLoadStats(lrsResponseObserverCaptor.capture());
+    assertThat(logs).containsExactly("DEBUG: LRS stream closed, backoff in 0 second(s)",
```
Comment (dapengzhang0) [inline xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: You don't need this either. This is just verifying `handleStreamClosed()`.  Stream is closed for sure. `assertThat(lrsRequestObservers).hasSize(1)` is sufficient to verify that backoff is 0 and retry happens.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 1 above)
Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Those logs are valuable for tracking retries and backoffs. That's why I verifies them extensively in the other retry-related test case (`lrsStreamClosedAndRetried`), as that tests the whole bunch of retry behaviors. For this one, we concern more about load reporting back to normal after retry. Maybe it's not that critical for logs, but I would think it's not bad to keep them as there are valuable logging events involved.

Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: I agree logs are valuable for tracking. But verifying logs in tests is kind of cheating, it's not testing real behavior. There are multiple `containsExactly()`, so if you add any other new logs you have to change every test.

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Removed.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 2 above)
Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Removed.

**Pull request merged**